nengo/learning_rules.py

```diff
@@ -315,6 +320,43 @@ def __init__(self, post_tau=0.005, learning_rate=1e-2):
         super(Voja, self).__init__(learning_rate, size_in=1)
 
 
+class GenericRule(LearningRuleType):
+    """Learning rule implemented by generic Python function.
+
+    Parameters
+    ----------
+    function : callable
+        Accepts previous values of learned array and a data vector
+        as input, outputs a delta for the learned array.
+
+        Optionally the function may also accept a `params` argument, in which
+        case `model.params` will be passed to the function as well (can be used
+        to look up things like encoder values).
```

Comment:
Yeah, it's awaiting a more permanent solution for how to access properties of the built model.  That's why I have it kind of hidden (optional, and turned off by default).  So it's there if someone needs it, but presumably if they understand what `model.params` is and how to use it, they will also understand the limitations.